sdp/processors/datasets/masc/aggregate_segments.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import logging
+from pydub import AudioSegment
+from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
+from sdp.processors.datasets.masc.utils import get_audio_segment
+
+class AggregateSegments(BaseParallelProcessor):
+    """
+    Aggregates short segments into segments with duration not longer than `max_duration`.
```

Comment:
Maybe explain briefly how it's a simple algorithm going from left to right and joining future segments to the current one as long as the total length doesn't exceed max_duration?